CONTRIBUTING.rst

```diff
@@ -75,15 +80,43 @@ Check the commits you have on your branch and then push them to your forked repo
+
+**Step 10**: when your pull request gets approved by at least one reviewer, you must squash your commits according to what has been explained above. In short:
 
-**Step 9**: once your pull request is ready to be reviewed, use the `Reviewers` option to ask a specific person(s) to review it. Watch your pull request for comments and feedback.
-If further changes are required to your pull request, you might want to provide them in a separate commit, making it easier to review only the latest difference.
+- logical and algorithmical changes go in the first commit;
+- while unit tests, pylint reformat, pep conventions, aesthetic minor improves go in the second commit.
+Make sure to provide meaningful - and short - commit messages, since they are used in our software release notes
+If you need help squashing your commits, please have a look at this short and clear document `this <https://steveklabnik.com/writing/how-to-squash-commits-in-a-github-pull-request>`_.
 
-**Step 10**: when your pull request gets approved by at least one reviewer, you must squash your commits accordingly in order to keep a clean commit history (remember, if you need to update both src/ and test/ files, then you need to squash them into the correct commits).
-If you need further instructions for squashing your commits, please check `this <https://steveklabnik.com/writing/how-to-squash-commits-in-a-github-pull-request>`_ quick and simple document.
+Timeframe expectations
+-----------------------
+People involved in the GH issues and pull requests should try commit to the following actions and their timeframes, such as:
+
+- GH issues: tickets inactive for more than 3 years will be candidates to be closed out without any development.
+- PR review: reviewers have 3 business days to provide feedback. If changes are too deep and/or complex, even a partial review is better than no review.
+- PR follow up: we ask the developer/contributor to follow up on any required changes and/or questions with a one month time period. Otherwise the team might consider it no longer relevant and it could become eligible to be closed out.
+
+Eventually we should integrate GH bots to start taking automated decisions based on some of these time period parameters.
+
+Useful pull request labels
+--------------------------
+
+The WMCore project has many labels, however, here is a list of the most important labels for pull request contributions:
+
+- ``Do not merge yet``: if changes are not fully read from your side, e.g. missing some extra validation, you might want to label your PR with this.
+- ``Work in progress``: if you created a pull request for a development that you know it's still unfinished, please use this label.
+- ``One approval required``: for somehow simple changes, or changes that are quite specific to a given WMCore service.
+- ``Two approvals required``: for more complex changes; or changes that are more intrusive and need special attention.
```

Comment:
Adding those two labels is really good for optimizing PR/review turnaround time.